mteb/cli.py

```diff
@@ -251,6 +253,56 @@ def add_run_parser(subparsers) -> None:
     parser.set_defaults(func=run)
 
 
+def potentially_add_cqadupstack_to_results(results: list[mteb.MTEBResults]) -> None:
+    """If all CQADupstack tasks are present in the results, combine them into a single CQADupstackRetrieval task and add it to the results."""
+    TASK_LIST_CQA = set(
+        [
+            "CQADupstackAndroidRetrieval",
+            "CQADupstackEnglishRetrieval",
+            "CQADupstackGamingRetrieval",
+            "CQADupstackGisRetrieval",
+            "CQADupstackMathematicaRetrieval",
+            "CQADupstackPhysicsRetrieval",
+            "CQADupstackProgrammersRetrieval",
+            "CQADupstackStatsRetrieval",
+            "CQADupstackTexRetrieval",
+            "CQADupstackUnixRetrieval",
+            "CQADupstackWebmastersRetrieval",
+            "CQADupstackWordpressRetrieval",
+        ]
+    )
+
+    task_names = {result.task_name for result in results}
+
+    if not TASK_LIST_CQA.issubset(task_names):
+        return None
+    cqa_results = [result for result in results if result.task_name in TASK_LIST_CQA]
+
+    evaluation_time = sum([result.evaluation_time for result in cqa_results])
+    main_scores = [r.get_score(splits=["test"]) for r in cqa_results]
+    main_score = float(sum(main_scores) / len(main_scores))
+    scores = {
+        "test": [
+            {
+                "main_score": main_score,
+                "ndcg_at_10": main_score,
+                "hf_subset": "default",
+                "languages": ["eng_Latn"],
+            }
+        ]
+    }
+
+    result = mteb.MTEBResults(
+        task_name="CQADupstackRetrieval",
+        dataset_revision="CQADupstackRetrieval is a combined dataset",
+        mteb_version="NA",
+        scores=scores,
+        evaluation_time=evaluation_time,
+        kg_co2_emissions=None,
```

Comment:
If we sum up the eval times, shouldn't we sum up the kg_co2 emissions too?

Reply:
It is not added to the markdown so I just avoided adding it.